services/diagnostic/sessions.go

```diff
@@ -0,0 +1,172 @@
+package diagnostic
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/kapacitor/uuid"
+)
+
+type SessionsStore interface {
+	Create(w http.ResponseWriter, contentType string, tags []tag) *Session
+	Delete(s *Session) error
+	Each(func(*Session))
+}
+
+type sessionsStore struct {
+	mu       sync.RWMutex
+	sessions map[uuid.UUID]*Session
+}
+
+func (kv *sessionsStore) Create(w http.ResponseWriter, contentType string, tags []tag) *Session {
+	kv.mu.Lock()
+	defer kv.mu.Unlock()
+
+	flusher, ok := w.(http.Flusher)
+	var wf WriteFlusher = &noopWriteFlusher{w: w}
```

Comment:
You could put this creation of the noopWriteFlusher in the else block. It would save an allocation :)